### PR TITLE
Mesh improvements

### PR DESCRIPTION
**Pull request by philippjfr**
Improvements for handling QuadMesh and TriMesh. Both @jbednar and I have been writing up documentation on dealing with meshes of different kinds and in doing so both of us hit some things that make it awkward to work with TriMeshes in particular at the moment. In particular in order to pass in a set of vertices with associated values you previously would have to construct a set of Nodes or Points yourself and pass those into the ``TriMesh`` constructor. With this change you can now pass in a pandas DataFrame of vertices and it will correctly infer the value dimension.

Secondly this PR refactors the regrid operation in such a way that a bunch of the code can be reused by regridding operations in other packages (specifically xESMF based regridding in GeoViews).

Finally this PR changes the way bokeh QuadMesh plots are created avoiding binary serialization. Sending a bunch of small array chunks seems to confuse bokeh and cause errors and actually adds more overhead than simply sending the data without binary transfer.

### CODE REVIEW

**Comment by philippjfr** [comment]
Ready for review.

**Comment by jbednar** [comment]
Looks good to me, thanks!

File: holoviews/plotting/bokeh/raster.py [hunk 1]
```diff
@@ -197,8 +197,8 @@ def get_data(self, element, ranges, style):
             XS, YS = [], []
             for x, y, zval in zip(X, Y, zvals):
                 if np.isfinite(zval):
-                    XS.append(x)
-                    YS.append(y)
+                    XS.append(list(x[:-1]))
+                    YS.append(list(y[:-1]))
```
Comment (jlstevens) [inline holoviews/plotting/bokeh/raster.py]: I assume there is a good reason though it seems odd to me you are slicing off the last element...

File: tests/testgraphelement.py [hunk 2]
```diff
@@ -130,13 +133,15 @@ def test_graph_redim_nodes(self):
 class TriMeshTests(ComparisonTestCase):
 
     def setUp(self):
-        self.nodes = [(0, 0, 0), (0.5, 1, 1), (1., 0, 2), (1.5, 1, 3)]
+        self.nodes = [(0, 0, 0), (0.5, 1, 1), (1., 0, 2), (1.5, 1, 4)]
```
Comment (jlstevens) [inline tests/testgraphelement.py]: The last value wasn't tested before?

**Comment by jlstevens** [comment]
Made two minor comments but otherwise it looks good to me!

File: holoviews/plotting/bokeh/raster.py (hunk 1 above)
Comment (philippjfr) [reply holoviews/plotting/bokeh/raster.py]: When drawing a polygon it automatically connects the first and last point, so the last point for each quad geometry is superfluous.

Comment (jlstevens) [reply holoviews/plotting/bokeh/raster.py]: Ok, that does make sense. Thanks!

File: tests/testgraphelement.py (hunk 2 above)
Comment (philippjfr) [reply tests/testgraphelement.py]: It was, but it wasn't a very good test. I've adjusted the test so it properly tests how the Graph index dimension is handled.

**Comment by jlstevens** [comment]
I'll merge once the tests are green.

**Comment by philippjfr** [comment]
Seems to be transients, I've restarted them.

**Pull request merged**